src/Nethermind/Nethermind.Blockchain/ChainHeadInfoProvider.cs

```diff
@@ -51,6 +58,24 @@ public ChainHeadInfoProvider(IChainHeadSpecProvider specProvider, IBlockTree blo
 
         public UInt256 CurrentFeePerBlobGas { get; internal set; }
 
+        private bool _hasSynced;
+        public bool IsSyncing
+        {
+            get
+            {
+                if (_hasSynced)
+                {
+                    return false;
+                }
+
+                long bestSuggestedNumber = _blockTree.FindBestSuggestedHeader()?.Number ?? 0;
+                long headNumberOrZero = _blockTree.Head?.Number ?? 0;
+                bool isSyncing = bestSuggestedNumber == 0 || bestSuggestedNumber > headNumberOrZero + 8;
```

Comment:
Can you make this extension method on block tree? I think I've seen this or similar logic in few other places before, can we unify?

Reply:
Done